smithy-aws-traits/src/main/java/software/amazon/smithy/aws/traits/tagging/TaggableResourceValidator.java

```diff
@@ -40,13 +40,13 @@
 public final class TaggableResourceValidator extends AbstractValidator {
     @Override
     public List<ValidationEvent> validate(Model model) {
-        List<ValidationEvent> events = new LinkedList<>();
+        List<ValidationEvent> events = new ArrayList<>();
         TopDownIndex topDownIndex = TopDownIndex.of(model);
         AwsTagIndex tagIndex = AwsTagIndex.of(model);
         for (ServiceShape service : model.getServiceShapes()) {
             for (ResourceShape resource : topDownIndex.getContainedResources(service)) {
                 boolean resourceLikelyTaggable = false;
-                if (resource.hasTrait(TaggableTrait.class)) {
+                if (resource.hasTrait(TaggableTrait.class) && service.getResources().contains(resource.getId())) {
                     events.addAll(validateResource(model, resource, service, tagIndex));
```

Comment:
The error messages in `validateResource` that are related to the service's tags should include the ID of the service shape to aid in debugging.